e2e/test/iothub/service/ModulesClientE2ETests.cs

```diff
@@ -65,15 +73,16 @@ public async Task ModulesClient_GetModulesOnDevice()
                     .Select(module => module.Id)
                     .ToList();
 
-                Assert.AreEqual(moduleCount, moduleIdsOnDevice.Count);
+                moduleIdsOnDevice.Count.Should().Be(moduleCount);
                 for (int i = 0; i < moduleCount; i++)
                 {
                     moduleIdsOnDevice.Should().Contain(testModuleIds[i]);
                 }
+
+                await Task.WhenAll(testModuleIds.Select(moduleId => serviceClient.Modules.DeleteAsync(testDeviceId, moduleId))).ConfigureAwait(false);
```

Comment:
why this is moved from finally block to here?

Reply:
Will fix/make irrelevant.